lib/wibox/widget/textclock.lua

```diff
@@ -55,6 +55,39 @@ function textclock.new(format, timeout, timezone)
         w._timer:again()
         return true -- Continue the timer
     end
+
+    --- Set the clock's format
+    -- @tparam string format The new time format.  This can contain pango
+    -- markup
+    function w.set_format(_, new_format)
```

Comment:
Please do not add functions in the constructor, add them outside and store the attributes in `self._private`

Also, use `@property` for properties and add an extra empty line under the documentation block.

Reply:
I thought as much.  Fair enough.  However, how should I then add the methods?  I mean, right now we're just returning a regular ol' textbox with a timer attached, not a textclock "class".

Reply:
Elv13's method seems to be:
```lua
local foo = {}
function foo:set_format(new_format)
    self._private.format = new_format
end

function constructor()
    local w = textbox()
    gears.table.crush(w, foo)
    return w
end
```